test/parallel/test-http2-compat-serverresponse-destroy.js

```diff
@@ -0,0 +1,95 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+// Check that destroying the Http2ServerResponse stream
+// produces the expected result, including the ability
+// to throw an error.
+
+const streamError = (err) => common.expectsError({
+  code: 'ERR_HTTP2_STREAM_ERROR',
+  type: Error,
+  message: 'Stream closed with error code 2'
+})(err);
+
+const server = http2.createServer(common.mustCall((req, res) => {
+  const path = req.headers[':path'];
+  let error;
+
+  if (path === '/error') {
+    req.once('error', (err) => assert.strictEqual(err, error));
+    res.once('error', (err) => assert.strictEqual(err, error));
```

Comment:
Is this behavior shared with HTTP1?  Can we check?

Reply:
Will confirm today.

Reply:
So in h1 it doesn't emit any server-side error events (on req, res or server itself) which seems a bit weird.

Reply:
This case is calling `destroy(err)` on the response, what does happen on H1?

In https://github.com/nodejs/node/pull/14991, errors from the underlining socket are not forwarded to req and res anymore.

Reply:
Actually, after testing this further, this code is clearly outdated. I was working on it at the same time as you were working on refactoring how the errors are emitted. Will revise.